CONTRIBUTING.md

```diff
@@ -0,0 +1,504 @@
+        }
+    ]
+}
+```
+
+**Policy 2**
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "mobileanalytics:PutEvents",
+                "cognito-sync:*"
+            ],
+            "Resource": [
+                "*"
+            ]
+        }
+    ]
+}
+```
+
+## Workflows
+
+### New Features or Bug Fixes
+
+The AWS SDK for iOS is mostly code generated off of models written in JSON and each time a new feature or new service at AWS launches, these models get updated and thus our SDK also gets updated. Beyond these low-level generated SDK's, we have built some higher level clients that predate the creation of [Amplify](https://github.com/aws-amplify/amplify-ios) (our high-level libraries built on top of the AWS SDK). These higher level clients include:
+
+- `AWSMobileClient`
+- `AWSCognitoAuth`
+- `AWSS3TransferUtility`
+- `AWSLexInteractionKit`
+- `AWSIoTDataManager`
+- `AWSPinpointTargetingClient`
+- `AWSPinpointAnalyticsClient`
+- `AWSPollySynthesizeSpeechURLBuilder`
+- `AWSTranscribeStreamingService`
+- `AWSMobileAnalyticsEventClient`
+
+You could go into any of these files and start adding new endpoints you see in the generated files to these higher level clients or fixing bugs as this is mostly likely where they would occur since it is not code generated code based off of models given to us by the service teams themselves.
+
+Once you have made your code change, make sure you can build and run both the Tests Folder and the UnitTests folder usually listed below the service folder. The Tests folder will be where the integration tests are and setting up your evironment to run the integration tests is detailed [above](#integration-testing-setup). Unit tests and/or integration tests are required for all features being added to the code-base. Updating existing tests is also required as well. Running tests in this project is the same as any other project in Xcode -> Set the scheme to be the service folder of the service you are editing. Then head to the tests tab on the left hand side of Xcode and run the tests you need to. If I was running Transcribe tests, Xcode would look like this:
+
+![image](readme-images/xcode-tests.png?raw=true)
+
+
+## Debugging
+
+### Running Cocoapods Locally
+
+Framework development is quite different from typical app development when it comes to debugging and being able to test your code. First you will want to create a new app that uses the pods you changed. For instance, if you changed something in the `AWSS3TransferUtility` to test to see if it works, you will need to create a new app and run `pod init` and edit the `Podfile` to use the local version of the SDK you have. This is done by editing the `Podfile` of the new application to take a path to the pod instead of a version number like below:
+
+```ruby
+
+target 'MySampleApp' do
+  # Comment the next line if you don't want to use dynamic frameworks
```

Comment:
we can remove this comment.... -- technically, they'd have to use_modular, not just comment it out.

Reply:
done